### PR TITLE
Fix Version Parsing & Test Activation

### PR DESCRIPTION
**Pull request by JL03-Yue**
#1661 

Includes an additional fix as well for when we added on startupfinished activation, which is elaborated upon in the commit history,

### CODE REVIEW

**Comment by nagilson** [comment]
@JL03-Yue  I think that other failure we were seeing in 'acquire' was due to a different problem, hopefully what I pushed just now will have fixed it though it may be a problem in another way, need to see how it works out.

Definitely go back to your original change and let's see if it works :)

**Comment by nagilson** [comment]
Hm, my fix did not work. There is something going on here outside your change that is broken. Ill look into this.

**Comment by nagilson** [comment]
Quick thought: I think this might be broken because of the way activation now works. If we activate the extension after startup, it could be that this was technically always broken because it tries to bundle the extension again when it runs the test, so it double registers things.

**Comment by JL03-Yue** [comment]
@nagilson The current code passed all tests and passed the scenarios on Red Hat. I wonder if I should rebase my commits? It is failing on packaging and publishing, which I'm looking into[smile]

**Comment by nagilson** [comment]
I think I fixed upstream.

**Review by nagilson — APPROVED** [review]

File: vscode-dotnet-runtime-library/src/Acquisition/VersionResolver.ts [hunk 1]
```diff
@@ -159,7 +159,7 @@ export class VersionResolver implements IVersionResolver {
         }
         Debugging.log(`Semver parsing passed: ${version}.`, this.context.eventStream);
 
-        if (version.split('.').length !== 2 || !parsedVer)
+        if (version.split('.').length !== 2 && version.split('.').length !== 3 || !parsedVer)
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/Acquisition/VersionResolver.ts:162]:
```suggestion
        if (!parsedVer || (version.split('.').length !== 2 && version.split('.').length !== 3))
```

**Review by nagilson — APPROVED** [review]

**Pull request merged**